### DISCUSSION TITLE
Consider adding a section dedicated to how to override `which-key` defaults, including overriding menu actions and titles

### DISCUSSION THREAD

**Discussion by macintacos** [LazyVim/LazyVim#3598] [Ideas]
`which-key` feels like its own "thing", similar to LSP keymaps, so I think that it needs its own section in the documentation. I could get over the quirks around only setting mappings in `keymap.lua` and handling LSP keybindings another way, but `which-key` feels like it could really use some clarification.

For example: I've searched through a lot of discussions here (+ the documentation of course) and it's still really unintuitive to figure out how to override the `<leader>h` action, so that it is a `+help` menu with my own keymaps. The only thing that worked was to delete the map _after_ `wk.register(opts.defaults)` was called - then the menu's title _and_ actions showed up properly together. I came up with something like the following, which works the way that I want:

```lua
return {
  "folke/which-key.nvim",
  config = function(_, opts)
    local wk = require("which-key")
    local del = vim.keymap.del

    wk.setup(opts)
    wk.register(opts.defaults)

    -- Delete top-level menu items here that you want to get overridden
    del("n", "<leader>h")

    wk.register({
    -- ...
```

But it's also inconsistent. For example, trying to call `del("n", "<leader>|")` to delete the top-level `<leader>|` action right after the `del("n", "<leader>h")` above doesn't work, and I get an error saying that that mapping doesn't exist - why?

It would be a great improvement to the documentation to point out the proper way to "override" every `which-key` mapping and menu that you get from LazyVim in the Keymaps section of the documentation, since that feels like the most intuitive place to put that information.

**Comment by dpetka2001** [comment]
`<leader>h` comes from the harpoon Extra. So, just change the mapping there to something else if you want to use it as a group name for which-key.

**Reply by macintacos** [reply]
Appreciate the pointer!

That almost makes my point though - I shouldn't have to figure that out on my own or by opening a discussion here, it should be documented and consistent how to override `which-key` mappings, considering that it's one of the major things users might want to reconfigure, just like regular keymaps. Hoping that this could be made easier in the future, because right now it's a bit inconsistent and hard to figure out.
